rasa/core/channels/channel.py

```diff
@@ -393,11 +393,21 @@ async def send_text_with_buttons(
         buttons: List[Dict[Text, Any]],
         **kwargs: Any,
     ) -> None:
-        await self._persist_message(
-            self._message(recipient_id, text=text, buttons=buttons)
-        )
+        """Sends text message with buttons."""
+        message_parts: List[Text] = text.strip().split("\n\n")
+        last_message_id: int = len(message_parts)
+
+        for i, message_part in enumerate(message_parts, start=1):
```

Comment:
Nit: Give the loop index a slightly more descriptive name like `idx` or `index`.